packages/api/amplify_api/test/dart_graphql_test.dart

```diff
@@ -0,0 +1,147 @@
+import 'dart:convert';
+
+import 'package:amplify_api/amplify_api.dart';
+import 'package:amplify_api/src/api_plugin_impl.dart';
+import 'package:amplify_core/amplify_core.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:http/http.dart' as http;
+import 'package:http/testing.dart';
+
+import 'test_data/fake_amplify_configuration.dart';
+
+// Success Mocks
+const _expectedQuerySuccessResponseBody = {
+  'data': {
+    'listBlogs': {
+      'items': [
+        {
+          'id': 'TEST_ID',
+          'name': 'Test App Blog',
+          'createdAt': '2022-06-28T17:36:52.460Z'
+        }
+      ]
+    }
+  }
+};
+const _expectedMutateSuccessResponseBody = {
+  'data': {
+    'createBlog': {
+      'id': 'TEST_ID',
+      'name': 'Test App Blog',
+      'createdAt': '2022-07-06T18:42:26.126Z'
+    }
+  }
+};
+
+// Error Mocks
+const _errorMessage = 'Not Authorized';
+const _errorLocations = [
+  {'line': 2, 'column': 3},
+  {'line': 4, 'column': 5}
+];
+const _errorPath = ['a', 1, 'b'];
+const _errorExtensions = {
+  'a': 'blah',
+  'b': {'c': 'd'}
+};
+const _expectedErrorResponseBody = {
+  'data': null,
+  'errors': [
+    {
+      'message': _errorMessage,
+      'locations': _errorLocations,
+      'path': _errorPath,
+      'extensions': _errorExtensions,
+    },
+  ]
+};
+
+class MockAmplifyAPI extends AmplifyAPIDart {
+  @override
+  http.Client getGraphQLClient({String? apiName}) =>
```

Comment:
nice job with this mocking logic. I think this will change as more cases get covered, or get copied/extended but this is a nice start.